cmd/snap/cmd_run_test.go

```diff
@@ -1931,6 +1941,34 @@ func (s *RunSuite) TestWaitWhileInhibitedGraphicalSessionFlowError(c *check.C) {
 	c.Assert(snaprun.WaitWhileInhibited("some-snap"), check.ErrorMatches, "boom")
 }
 
+func (s *RunSuite) TestWaitWhileInhibitedGraphicalSessionFlowErrorOnFinish(c *check.C) {
+	restoreIsGraphicalSession := snaprun.MockIsGraphicalSession(true)
+	defer restoreIsGraphicalSession()
+
+	restorePendingRefreshNotification := snaprun.MockPendingRefreshNotification(func(refreshInfo *usersessionclient.PendingSnapRefreshInfo) error {
+		return nil
+	})
+	defer restorePendingRefreshNotification()
+
+	restoreFinishRefreshNotification := snaprun.MockFinishRefreshNotification(func(refreshInfo *usersessionclient.FinishedSnapRefreshInfo) error {
```

Comment:
Is it possible to mock some calls deeper in the stack eg in usersessionclient?

Reply:
I think that the current approach is the best one for unit testing. Testing the actual behaviour is better done in component tests (which we don't have) or integration tests (spread).

Reply:
I was hoping for something more elaborate, but it's clearly consistent with other unit tests in this package, so I guess it's fine.